controller/rest/rest.go

```diff
@@ -1425,6 +1436,23 @@ func InitContext(ctx *Context) {
 	if err := initJWTSignKey(); err != nil {
 		log.WithError(err).Error("failed to initialize JWT sign key.")
 	}
+
+	searchRegistries = utils.NewSet()
+
+	for _, reg := range strings.Split(ctx.SearchRegistries, ",") {
+		if parsedReg, err := url.Parse(reg); parsedReg.Host != "" {
```

Comment:
When url.Parse() returns non-nil err, parsedReg is nil and parsedReg.Host triggers panic.
err should be nil-checked first and then parsedReg.Host can be accessed

Reply:
Good catch! I have swapped the order of the if statements to make sure the error checking happens first